### ISSUE TITLE
Getting an invalid key from a map will add that key to the map.

### ISSUE THREAD

**Issue by marekmaskarinec** [vtereshkov/umka-lang#336]
```
fn main() {
	a := map[str]int {
		"x": 1,
		"y": 2
	}

	printf("a: %v\n", a)
	b := a["z"]
	printf("b: %v\na: %v\n", b, a)
}
```

Expected result:

```
a: {"y": 2 "x": 1}
b: 0
a: {"y": 2 "x": 1}
```

Actual result:

```
a: {"y": 2 "x": 1}
b: 0
a: {"y": 2 "z": 0 "x": 1}
```

**Comment by vtereshkov** [comment]
@marekmaskarinec Yes, this is a known problem (see #170). In fact, Umka's maps behave like in C++: any access to a map item either fetches an existing item, or creates a new one. For now, please use `validkey()`.

**Comment by vtereshkov** [comment]
Closed as duplicate of #170.
